nimterop/getters.nim

```diff
@@ -634,6 +644,109 @@ proc getNameKind*(name: string): tuple[name: string, kind: Kind, recursive: bool
   if result.kind != exactlyOne:
     result.name = result.name[0 .. ^2]
 
+proc getCommentsStr*(gState: State, commentNodes: seq[TSNode]): string =
+  ## Generate a comment from a set of comment nodes. Comment is guaranteed
+  ## to be able to be rendered using nim doc
+  if commentNodes.len > 0:
+    result = "::"
+    for commentNode in commentNodes:
+      result &= "\n  " & gState.getNodeVal(commentNode).strip()
+
+    result = result.replace(re" *(//|/\*\*|\*\*/|/\*|\*/|\*)", "")
+    result = result.multiReplace([("\n", "\n  "), ("`", "")]).strip()
+
+proc getCommentNodes*(gState: State, node: TSNode, maxSearch=1): seq[TSNode] =
```

Comment:
Not urgent for this PR but please document what the tree-sitter AST looks like and what you are searching for so that it is easier to understand the algorithm.

Reply:
Alright, will do!